src/main/java/org/testng/internal/thread/graph/GraphThreadPoolExecutor.java

```diff
@@ -26,18 +27,16 @@
   /** Set to true if you want to generate GraphViz graphs */
   private static final boolean DOT_FILES = false;
 
-  private DynamicGraph<T> m_graph;
+  private final DynamicGraph<T> m_graph;
   private List<Runnable> m_activeRunnables = Collections.synchronizedList(Lists.<Runnable>newArrayList());
   private IThreadWorkerFactory<T> m_factory;
   private List<String> m_dotFiles = Lists.newArrayList();
-  private int m_threadCount;
+  private boolean useLatches;
 
   public GraphThreadPoolExecutor(DynamicGraph<T> graph, IThreadWorkerFactory<T> factory, int corePoolSize,
       int maximumPoolSize, long keepAliveTime, TimeUnit unit, BlockingQueue<Runnable> workQueue) {
-    super(corePoolSize, maximumPoolSize, keepAliveTime, unit, workQueue
-        /* , new TestNGThreadPoolFactory() */);
```

Comment:
I dug and found how to fix the `TestNGThreadPoolFactory` issue: just change `TestNG` by something else. For example, "testng", "TestNg" or "testNG" are working. But "TestNG" anywhere in the thread name is failing.

I'd love to understand why "TestNG" is failing. 
@cbeust Any idea?